dejavu-fonts: 2.35 -> 2.37, simplify build process

cmfwyp opened the pull request:
###### Motivation for this change

Release 2.36 adds and fixes a number of glyphes, and adds the Tex Gyre DejaVu Math font, a companion font to DejaVu Serif for typesetting mathematics.

Release 2.37 fixes an issue with condensed typefaces.

Unicode data and fc-lang are only necessary to generate the status files and coverage information, which are not used, or indeed even generated with full-ttf.
###### Things done
- [x] Tested using `nix-build --attr pkgs.dejavu_fonts default.nix`, with nix.useSandbox
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
dejavu-fonts: 2.35 -> 2.37

Release 2.36 adds and fixes a number of glyphes, and adds the
Tex Gyre DejaVu Math font, a companion font to DejaVu Serif for
typesetting mathematics.

Release 2.37 fixes an issue with condensed typefaces.

The sources are now fetched from GitHub, since the development was
moved from SourceForge to GitHub.

rycee commented on pkgs/data/fonts/dejavu-fonts/default.nix:
```diff
@@ -1,36 +1,23 @@
+  src = fetchFromGitHub {
+    owner = "dejavu-fonts";
+    repo = "dejavu-fonts";
+    rev = "0eda8a319c08835009849583cd090bb5b141ce25";
```
I think it would be more intuitive to have
```
    rev = "version_2_37";
```
here. Or perhaps even
```
    rev = "version_${stdenv.lib.replaceStrings ["."] ["_"] version}";
```

cmfwyp replied:
All of the version tags since version_0_0 follow that format, so the second option seems best.

rycee commented on pkgs/data/fonts/dejavu-fonts/default.nix:
```diff
@@ -1,36 +1,23 @@
 
   buildFlags = "full-ttf";
 
   preBuild = ''
```
While cleaning up the build I think it might also be worth to replace these `sed` calls by simply
```
  preBuild = "patchShebangs scripts";
```

Commit:
dejavu-fonts: simplify build process

Unicode data and fc-lang are only necessary to generate the status
files and coverage information, which are not used, or indeed even
generated with full-ttf.

cmfwyp commented:
Done.

rycee commented:
Looking good, thanks! Merged in commit 41b8c6d5a9523598d33e81e0ce7a70d02d748bb3.

vcunat commented:
Does anyone have an idea why we build the font instead of using binary `*.ttf` from upstream directly?

Doing that by default would help me to break a cycle when solving https://github.com/NixOS/nixpkgs/issues/17571.

7c6f434c commented:
>Does anyone have an idea why we build the font instead of using binary `*.ttf` from upstream directly?

Open source, they said. I saw a nice source package and built it — fonts
contain enough magic to be considered software and be philosophically 
worth a source build when it's reasonably cheap.

Let's just say I thought in terms of adding a «Bunch of Unicode fonts»
(actual commit message), not the default system font with closure 
considerations for the minimal graphical install.

>Doing that by default would help me to break a cycle when solving https://github.com/NixOS/nixpkgs/issues/17571.

Well, if there is a good reason to use prebuilt upstream package, then
it's an understandable change.

Never thought I would participate in a discussion of fonts from 
_bootstrapping_ point of view, to be honest… (not a complaint, just an
observation)

vcunat commented:
In the end, I think I'll manage to force the non-gui version of `fontforge` build without any fontconfig, thus cutting the cycle.
